src/server/game/Entities/Creature/Creature.h

```diff
@@ -659,7 +659,7 @@ class Creature : public Unit, public GridObject<Creature>, public MapObject
 
         // Handling caster facing during spellcast
         void SetTarget(ObjectGuid guid) override;
-        void FocusTarget(Spell const* focusSpell, WorldObject const* target);
+        void FocusTarget(Spell const* focusSpell, ObjectGuid targetGuid);
```

Comment:
Updated, instead of calling focusTarget with no target, I removed the creatures target when it has no target or when target is self